debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -148,7 +149,7 @@ private void takeSnapshot(Consumer<SourceRecord> consumer) {
             // we're locking in SHARE UPDATE EXCLUSIVE MODE to avoid concurrent schema changes while we're taking the snapshot
             // this does not prevent writes to the table, but prevents changes to the table's schema....
             schema.tables().forEach(tableId -> statements.append("LOCK TABLE ")
-                                                         .append(tableId.toString())
+                                                         .append(tableId.toQuotedId(PostgresTableIdTransformer.INSTANCE))
```

Comment:
Always quoting identifiers may be unnecessary but should have no side effects.  My thinking was it would be more intensive to work out whether an identifier needs escaping.

It is possible to put the transformer within TableId but given not all DB engines require transformation I thought to make it external.  Also, TableId is currently marked as final making so making it an extension point would be difficult and most methods are static which also makes storing state tougher (I had tried).